### DISCUSSION TITLE
Converting database fails

### DISCUSSION THREAD

**Discussion by jiriks74** [nextcloud/all-in-one#4743] [Questions] [answered]
I wanted to convert from the former Docker image to AIO. I used the documentation, spun up a temporary PostgreSQL server and ran `php occ db:convert-type --all-apps --password "$PG_PASSWORD" pgsql "$PG_USER" server_ip "$PG_DATABASE"`.

But I get errors converting multiple tables:

![image](https://github.com/nextcloud/all-in-one/assets/54378412/b7e626fc-01b2-48e6-977d-dfb0c168a17b)

The tables do not have any duplicates in them - I looked using DBeaver and there's nothing that is duplicated.

I was able to solve this issue by exporting it to a second database, deleting it, and restoring it from the export. Doing a database dump and loading it again doesn't work. I cannot do this for all the tables as the program is too unstable to do it and I didn't find anything that has these capabilities and would be stable.

I also noticed that when I basically recreate the table the conversion tool writes all the entries into the table so that is probably why I'm getting that error.

**Comment by jiriks74** [comment]
I managed to recover 99% of the database. I lost a few rows from something I don't consider important (like the broken News app) and I got the conversion running. But I ran into another issue:

![image](https://github.com/nextcloud/all-in-one/assets/54378412/4437df5e-cfd7-4730-af91-de0f4cc13de4)

**Comment by szaimen** [comment, marked as answer]
Hi, I fear you are running into this: https://github.com/nextcloud/server/issues/45257

**Reply by jiriks74** [reply]
It sure does but the process never got to the part where it dropped the tables from the original database. It just duplicated everything in it and then failed with `Operation xy is not supported by platform`. Maybe MariaDB doesn't support some thing that MySQL does and that prevented it from dropping the tables. I'll try the modifications to make it work properly (as the **official** *unofficial* docker image is quite bad with being up to date...).
